diff --git a/usr/share/lipstick-jolla-home-qt5/compositor.qml b/usr/share/lipstick-jolla-home-qt5/compositor.qml
index 1a8e7a5..d8a3153 100644
--- a/usr/share/lipstick-jolla-home-qt5/compositor.qml
+++ b/usr/share/lipstick-jolla-home-qt5/compositor.qml
@@ -597,6 +597,7 @@ Compositor {
                 peekFilter.topEnabled: Desktop.settings.close_from_top
 
                 onAboutToShowWindow: {
+                    root.launcherHinting = false
                     root._homeLayerOnPeek = homeLayerItem.switcher
                     homeLayerItem.setCurrentItem(homeLayerItem.switcher, root.homeVisible)
                     if (!lockScreenLayer.active) {
diff --git a/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml b/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml
index 26e0dfd..94c81e8 100644
--- a/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml
@@ -224,7 +224,7 @@ Layer {
             id: hintShowAnimation
             target: edgeLayer
             property: "_hintProgress"
-            duration: 300
+            duration: 200
             to: Theme.iconSizeLauncher * 2
             easing.type: Easing.InOutQuad
         }
@@ -232,7 +232,7 @@ Layer {
         NumberAnimation {
             target: edgeLayer
             property: "_hintProgress"
-            duration: 300
+            duration: 200
             to: 0
             easing.type: Easing.InOutQuad
         }
diff --git a/usr/share/lipstick-jolla-home-qt5/main.qml b/usr/share/lipstick-jolla-home-qt5/main.qml
index dae52cc..9b9c1b3 100644
--- a/usr/share/lipstick-jolla-home-qt5/main.qml
+++ b/usr/share/lipstick-jolla-home-qt5/main.qml
@@ -100,22 +100,9 @@ ApplicationWindow {
             onTriggered: Lipstick.compositor.launcherHinting = true
         }
 
-        MouseArea {
+        Switcher {
+            id: switcher
             anchors.fill: parent
-
-            enabled: switcher.count == 0 && Lipstick.compositor.systemInitComplete
-            drag.filterChildren: true
-
-            onClicked: {
-                if (enabled) {
-                    Lipstick.compositor.launcherHinting = true
-                }
-            }
-
-            Switcher {
-                id: switcher
-                anchors.fill: parent
-            }
         }
 
         Binding {
diff --git a/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml b/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
index 410f9ab..495f4b7 100644
--- a/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
+++ b/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
@@ -52,6 +52,8 @@ SilicaFlickable {
     onCloseApplicationEnabledChanged: {
         minimizeLaunchingWindows()
         switcherGrid.updateColumns()
+        if (closeApplicationEnabled)
+            Lipstick.compositor.launcherHinting = false
     }
 
     function minimizeLaunchingWindows() {
@@ -553,6 +555,10 @@ SilicaFlickable {
 
             if (switcherRoot.closeApplicationEnabled && !switcherRoot.housekeepingMenuActive)
                 switcherRoot.closeApplicationEnabled = false;
+            else if (Lipstick.compositor.systemInitComplete) {
+                switcherRoot.minimizeLaunchingWindows()
+                Lipstick.compositor.launcherHinting = true
+            }
         }
 
         function adjustOOMScores() {
