diff --git a/usr/share/lipstick-jolla-home-qt5/compositor.qml b/usr/share/lipstick-jolla-home-qt5/compositor.qml
index 2a3c9cc..18abbb3 100644
--- a/usr/share/lipstick-jolla-home-qt5/compositor.qml
+++ b/usr/share/lipstick-jolla-home-qt5/compositor.qml
@@ -688,6 +688,7 @@ Compositor {
                 }
 
                 onAboutToShowWindow: {
+                    root.launcherHinting = false
                     root.showApplicationOverLockscreen = Desktop.startupWizardRunning
                     homeLayerItem.lastActiveLayer = homeLayerItem.switcher
                     homeLayerItem.setCurrentItem(homeLayerItem.switcher, root.homeVisible)
diff --git a/usr/share/lipstick-jolla-home-qt5/main.qml b/usr/share/lipstick-jolla-home-qt5/main.qml
index 3e570ca..87a9cd7 100644
--- a/usr/share/lipstick-jolla-home-qt5/main.qml
+++ b/usr/share/lipstick-jolla-home-qt5/main.qml
@@ -105,15 +105,9 @@ ApplicationWindow {
             objectName: "main"
             anchors.fill: parent
 
-            enabled: switcher.count == 0 && Lipstick.compositor.systemInitComplete
+            enabled: false
             drag.filterChildren: true
 
-            onClicked: {
-                if (enabled) {
-                    Lipstick.compositor.launcherHinting = true
-                }
-            }
-
             Switcher {
                 id: switcher
                 anchors.fill: parent
diff --git a/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml b/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
index 86a9624..798837b 100644
--- a/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
+++ b/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
@@ -53,6 +53,8 @@ SilicaFlickable {
     onCloseApplicationEnabledChanged: {
         minimizeLaunchingWindows()
         switcherGrid.updateColumns()
+        if (closeApplicationEnabled)
+            Lipstick.compositor.launcherHinting = false
     }
 
     function minimizeLaunchingWindows() {
@@ -588,6 +590,10 @@ SilicaFlickable {
 
             if (switcherRoot.closeApplicationEnabled && !switcherRoot.housekeepingMenuActive)
                 switcherRoot.closeApplicationEnabled = false;
+            else if (Lipstick.compositor.systemInitComplete) {
+                switcherRoot.minimizeLaunchingWindows()
+                Lipstick.compositor.launcherHinting = true
+            }
         }
 
         function adjustOOMScores() {
